PokeDemoTests/MockService.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  MockService.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 15/09/22.
+//
+
+import Foundation
+import Alamofire
+@testable import PokeDemo
+
+final class MockService: ServiceRepresentable {
+    
+    var sendRequestWithJSONIsCalledIndex = 0
+    var sendRequestWithJSONEndPoint: String = ""
```

Comment:
Create these properties read only. We don't want that in tests these values can be overwritten by someone. Its value must be changed only by the methods which are called internally for which we are writing tests

Reply:
suggestion - var sendRequestWithJSONCallCount = 0

Reply:
You need to create read only variables for method, parameter and header too. To test right values are passed for these as well or not.

Reply:
Covered in 9bbf1c08498fe5f761b27131dee55b2fd25072cd